polars/polars-time/src/chunkedarray/utf8/mod.rs

```diff
@@ -275,29 +277,51 @@ pub trait Utf8Methods: AsUtf8 {
                         if s.is_empty() {
                             return None;
                         }
-                        match NaiveDateTime::parse_from_str(s, fmt).map(func) {
-                            Ok(nd) => return Some(nd),
-                            Err(e) => {
-                                let e: ParseErrorByteCopy = e.into();
-                                match e.0 {
-                                    ParseErrorKind::TooLong => {
-                                        s = &s[..s.len() - 1];
-                                    }
-                                    _ => {
-                                        s = &s[i..];
+                        match tz_aware {
```

Comment:
nit: `if else` feels more ideomatic than matching booleans.

Reply:
yup, thanks - found a simpler way to write it anyway, assigning an intermediate part of the computation to `timestamp: i64` (this also helped notice that it was misnamed as `nd` (presumably "naive date") in one of the arms)